git/repo.go

```diff
@@ -345,3 +350,92 @@ func (repo *Repository) ObjectForPath(root Object, pathstr string) (Object, erro
 
 	return node, nil
 }
+
+// ParseCommitList executes a custom git log command of the specified branch of the
+// associated git repository and returns the resulting list of commits as an array.
+func (repo *Repository) ParseCommitList(branch string) ([]wire.CommitListItem, error) {
```

Comment:
Not sure `Parse` is the right name, it is not only parsing the commit list, but getting it via git.
The name for the `branch` parameter probably could be something like `rev`/`ref`/`refspec` or something, since that should equally work for tags and as (non-type-safe) string currently even for revision ranges.